src/ncit/index.js

```diff
@@ -119,60 +114,76 @@ const cleanRawRow = (rawRow) => {
         ),
         description: definition,
+        name: rawName.trim(),
         parents: (
             rawParents.split('|')
                 .map(parent => parent.trim())
                 .filter(parent => parent && !DEPRECATED.includes(parent))
                 .map(parent => parent.toLowerCase())
         ),
+        species: '',
+        synonyms: (rawSynonyms.split('|')
+            .map(s => s.trim())
+            .filter(s => s)),
     };
     const sourceId = id.toLowerCase().trim();
     const endpoint = pickEndpoint(semanticType, parentConcepts);
 
-    // use the synonym name if no name given
-    const synonyms = rawSynonyms.split('|')
-        .map(s => s.trim())
-        .filter(s => s);
-    let name = rawName.trim();
-
     // split up the name if it is a list
-    if (name && name.includes('|')) {
-        const names = name.split('|')
+    if (row.name && row.name.includes('|')) {
+        const names = row.name.split('|')
             .map(s => s.trim())
             .filter(s => s);
-        [name] = names;
-        synonyms.push(...names.slice(1));
+        [row.name] = names;
+        row.synonyms.push(...names.slice(1));
     }
 
+    const speciesMatch = (termName) => {
+        const m = /\b(murine|mouse|rat)\b/ig.exec(termName);
+
+        if (m) {
+            return m[1];
+        }
+        return '';
+    };
+
     // non-human concepts should use fuller name
-    if (!/\b(murine|mouse|rat)\b/.exec(name)) {
-        for (const synonym of synonyms) {
-            if (/\b(murine|mouse|rat)\b/.exec(synonym)) {
-                name = synonym;
+    if (!speciesMatch(row.name)) {
+        for (const synonym of row.synonyms) {
+            if (speciesMatch(synonym)) {
+                row.name = synonym;
+                [, row.species] = speciesMatch(synonym);
                 break;
             }
         }
+        const species = row.parents.map(speciesMatch).filter(s => s);
+
+        if (species.length) {
+            [row.species] = species;
+        }
+    } else {
+        row.species = speciesMatch(row.name);
     }
 
     // use the synonym name if no name given
-    if (!name) {
-        name = sourceId;
+    if (!row.name) {
+        row.name = sourceId;
     }
 
     const url = xmlTag.replace(/^</, '').replace(/>$/, '');
 
     // add the parents
     return {
         ...row,
-        displayName: name.toLowerCase() === sourceId.toLowerCase()
+        displayName: row.name.toLowerCase() === sourceId.toLowerCase()
             ? sourceId
-            : `${name} [${sourceId}]`,
+            : `${row.name} [${sourceId}]`,
         endpoint,
-        name: name.toLowerCase(),
+        name: row.name.toLowerCase(),
         sourceId,
-        synonyms: Array.from(new Set(synonyms))
+        synonyms: Array.from(new Set(row.synonyms))
             .map(s => s.toLowerCase())
-            .filter(s => s !== name.toLowerCase()),
+            .filter(s => s !== row.name.toLowerCase()),
         url,
```

Comment:
is there a reason for making name/species properties of row?

Reply:
This is done b/c they were originally properties of row, i was setting them to local variables, manipulating them and then re-setting them on row, It seemed unnecessarily confusing. this way I am sure there is only 1 value for a given property going on at 1 time